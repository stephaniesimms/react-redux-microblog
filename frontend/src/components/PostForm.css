.PostForm { 
  border: 1px solid #D3D3D3; 
  border-radius: 10px; 
  padding: 3em; 
  margin: auto;
}