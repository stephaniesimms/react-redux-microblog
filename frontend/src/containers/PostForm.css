.PostForm { 
  border: 2px solid rgba(0,0,0,.125); 
  border-radius: 10px; 
  padding: 10px; 
  margin: auto;
  max-width: 700px;
}