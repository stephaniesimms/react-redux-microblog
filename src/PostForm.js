import React, { Component } from 'react';
import Button from 'react-bootstrap/Button';
import Form from 'react-bootstrap/Form';
import uuid from 'uuid/v4';

class PostForm extends Component {
  constructor(props) {
    super(props)

    this.state = {
      title: '',
      description: '',
      body: ''
    };

    this.handleChange = this.handleChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
  }

  handleChange(evt) {
    this.setState({
      [evt.target.name]: evt.target.value
    });
  };

  async handleSubmit(evt) {
    evt.preventDefault();

    this.props.addPost({ ...this.state, id: uuid() });
    this.setState({
      title: '',
      description: '',
      body: ''
    });
  };

  render() {
    return (
      <div style={{ border: "2px solid rgba(0,0,0,.125)", borderRadius: "10px", padding: "10px", overflow: "auto" }}>
        <Form onSubmit={this.handleSubmit}>
          <Form.Group>
            <Form.Label>Title</Form.Label>
            <Form.Control id="title"
                          name="title"
                          onChange={this.handleChange}
                          value={this.state.title} 
                          size="md" 
                          type="text" />
          </Form.Group>
          <Form.Group>
          <Form.Label>Description</Form.Label>
            <Form.Control id="description"
                          name="description"
                          onChange={this.handleChange}
                          value={this.state.description}
                          size="md" 
                          type="text" /> 
          </Form.Group>
          <Form.Group>
          <Form.Label>Body</Form.Label>
            <Form.Control id="body"
                          name="body"
                          onChange={this.handleChange}
                          value={this.state.body} 
                          size="md" 
                          as="textarea" /> 
          </Form.Group>
          <Button type="submit" 
                  variant="outline-dark" 
                  size="md">
                    Post
          </Button>
        </Form>
      </div>
    );
  }
}

export default PostForm;