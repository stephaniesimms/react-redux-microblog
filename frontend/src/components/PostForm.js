import React, { Component } from 'react';
import { Button, Form } from 'react-bootstrap';
import './PostForm.css';
import { Link } from 'react-router-dom';

/** Show post form.
 *
 * Can be used for editing/adding --- it calls the parent with edit
 * or cancel actions.
 *
 */
class PostForm extends Component {
  static defaultProps = {
    post: { title: '', description: '', body: '' },
  };

  constructor(props) {
    super(props)
    this.state = {
      title: this.props.post.title,
      description: this.props.post.description,
      body: this.props.post.body,
    };
    this.handleChange = this.handleChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
  }

  handleChange(evt) {
    this.setState({
      [evt.target.name]: evt.target.value
    });
  };

  handleSubmit(evt) {
    evt.preventDefault();
    this.props.save(this.state)
  };

  render() {
    return (
      <div className='PostForm p-3'>
        <Form onSubmit={this.handleSubmit}>
          <Form.Group>
            <Form.Label>Title</Form.Label>
            <Form.Control id='PostForm-title'
              name='title'
              onChange={this.handleChange}
              value={this.state.title}
              type='text'
              alt='title' />
          </Form.Group>
          <Form.Group>
            <Form.Label>Description</Form.Label>
            <Form.Control id='PostForm-description'
              name='description'
              onChange={this.handleChange}
              value={this.state.description}
              type='text'
              alt='description' />
          </Form.Group>
          <Form.Group>
            <Form.Label>Body</Form.Label>
            <Form.Control id='PostForm-body'
              name='body'
              onChange={this.handleChange}
              value={this.state.body}
              as='textarea'
              alt='body'
              rows='10' />
          </Form.Group>
          <div className='row justify-content-end'>
            <Button className='mr-2'
              type='submit'
              variant='dark'>
              Post
          </Button>
            <Link to='/'>
              <Button className='mr-3' 
                variant='danger'>
                Cancel
            </Button>
            </Link>
          </div>
        </Form>
      </div>
    );
  }
}

export default PostForm;