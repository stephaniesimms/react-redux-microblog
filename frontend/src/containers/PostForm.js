import React, { Component } from 'react';
import Button from 'react-bootstrap/Button';
import Form from 'react-bootstrap/Form';
import Row from 'react-bootstrap/Row';
import Col from 'react-bootstrap/Col'
import './PostForm.css';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import { sendPostToAPI } from '../actions/actions';

// TODO: should PostForm be a component instead? 
// maybe refactor to follow solution
// otherwise update docstring

/** Show post form.
 *
 * Can be used for editing/adding --- it just calls the parent with edit
 * or cancel actions.
 *
 */
class PostForm extends Component {
  constructor(props) {
    super(props)
    this.state = {
      title: '',
      description: '',
      body: '',
    };
    this.handleChange = this.handleChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
  }

  componentDidMount() {
    if (this.props.post) {
      const { title, description, body } = this.props.post;

      this.setState({
        title,
        description,
        body,
      });
    }
  }

  handleChange(evt) {
    this.setState({
      [evt.target.name]: evt.target.value
    });
  };

  /** Checks the formType prop sent from Post component to decide 
   * whether to create a new post or edit an existing post
  */
  async handleSubmit(evt) {
    evt.preventDefault();

    if (this.props.formType === 'Edit') {
      this.props.updatePost({
        ...this.state,
        comments: this.props.post.comments
      });
    } else {
      const { title, description, body } = this.state;
      await this.props.sendPostToAPI({ title, description, body });

      this.props.history.push('/');
    }
  };

  render() {
    return (
      <div className='PostForm'>
        <h3>{this.props.formType}</h3>
        <Form onSubmit={this.handleSubmit}>
          <Form.Group as={Row}>
            <Form.Label column sm={2}>Title</Form.Label>
            <Col sm={10}>
              <Form.Control id='title'
                name='title'
                onChange={this.handleChange}
                value={this.state.title}
                size='md'
                type='text'
                alt='title' />
            </Col>
          </Form.Group>
          <Form.Group as={Row}>
            <Form.Label column sm={2}>Description</Form.Label>
            <Col sm={10}>
              <Form.Control id='description'
                name='description'
                onChange={this.handleChange}
                value={this.state.description}
                size='md'
                type='text'
                alt='description' />
            </Col>
          </Form.Group>
          <Form.Group as={Row}>
            <Form.Label column sm={2}>Body</Form.Label>
            <Col sm={10}>
              <Form.Control id='body'
                name='body'
                onChange={this.handleChange}
                value={this.state.body}
                size='md'
                as='textarea'
                alt='body' rows='5' />
            </Col>
          </Form.Group>
          <div className='row justify-content-end'>
          <Button className='mr-2' type='submit' variant='dark'>
            Post
          </Button>
          <Link to='/'>
              <Button className='mr-3' variant='danger'>
              Cancel
            </Button>
          </Link>
          </div>
        </Form>
      </div>
    );
  }
}

function mapStateToProps(state) {
  return { posts: state.posts };
}


export default connect(mapStateToProps, { sendPostToAPI })(PostForm);