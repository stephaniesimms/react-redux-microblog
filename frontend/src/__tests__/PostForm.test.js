import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import PostForm from '../components/PostForm';

it('renders without crashing', function () {
  shallow(<PostForm />);
});

it('matches snapshot', function () {
  let wrapper = shallow(<PostForm />);
  let serialized = toJson(wrapper);
  expect(serialized).toMatchSnapshot();
});