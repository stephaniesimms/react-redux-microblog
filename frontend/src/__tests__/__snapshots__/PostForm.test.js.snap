// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="PostForm p-3"
>
  <Form
    inline={false}
    onSubmit={[Function]}
  >
    <FormGroup>
      <FormLabel
        column={false}
        srOnly={false}
      >
        Title
      </FormLabel>
      <FormControl
        alt="title"
        id="PostForm-title"
        name="title"
        onChange={[Function]}
        type="text"
        value=""
      />
    </FormGroup>
    <FormGroup>
      <FormLabel
        column={false}
        srOnly={false}
      >
        Description
      </FormLabel>
      <FormControl
        alt="description"
        id="PostForm-description"
        name="description"
        onChange={[Function]}
        type="text"
        value=""
      />
    </FormGroup>
    <FormGroup>
      <FormLabel
        column={false}
        srOnly={false}
      >
        Body
      </FormLabel>
      <FormControl
        alt="body"
        as="textarea"
        id="PostForm-body"
        name="body"
        onChange={[Function]}
        rows="10"
        value=""
      />
    </FormGroup>
    <div
      className="row justify-content-end"
    >
      <Button
        active={false}
        className="mr-2"
        disabled={false}
        type="submit"
        variant="dark"
      >
        Post
      </Button>
      <Link
        to="/"
      >
        <Button
          active={false}
          className="mr-3"
          disabled={false}
          type="button"
          variant="danger"
        >
          Cancel
        </Button>
      </Link>
    </div>
  </Form>
</div>
`;
